### ISSUE TITLE
IGpioPin does not allow access to PWM functions

### ISSUE THREAD

**Issue by BryanCrotaz** [unosquare/wiringpi-dotnet#30]
`IGpioPin` should give access to all PWM functions.

In Workbench, the Servo demo should take an `IGpioPin` in the constructor, not a `GpioPin`.

**Comment by wezzix** [comment]
Try a cast. The readme gives the following example:
```
   var range = 100;
   var pin = (GpioPin)Pi.Gpio[BcmPin.Gpio24];
   pin.PinMode = GpioPinDriveMode.Output;
   pin.StartSoftPwm(0, range);
```

**Comment by BryanCrotaz** [comment]
Oh that's soooo dodgy! Assuming that the implementation of an interface is an exact class is ripe with error potential.

**Comment by wezzix** [comment]
Yes. I don't know why the maintainers designed it so. Maybe they'd like to chip in?

**Comment by BryanCrotaz** [comment]
I've submitted a few improvement PRs but they're withering on the vine

**Comment by geoperez** [comment]
We are looking for maintainers, you can apply at https://adoptoposs.org/p/67b02bee-a00e-446a-a321-c1a6449014bc
